nss-day-cohort-41/react-nutshell-the-macadamias
# Chat component

# Description
Created messages component adding a messages route/path in the nav bar. The messages page displays all the messages from the active user, friends of active user and others. The latest message appears last. Within each message card, the edit button is visible to the active user only and the add button is available only to non-friends of the active user. Additionally, there is a text input field at the bottom of the messages display to add new messages. 

Features:
- Post new message is complete
- Edit existing message is complete
- Add new friend from messages is still in development 

## Type of change

# Testing Instructions
1. Type new message in text field and click send to add new message, the messages display should update with new message. 2. Click on edit button for edit modal to appear, prior message should appear in text field, modify text as needed and click submit changes, the messages display should update with the modified message.

# Checklist:

# Sample JSON data to use:

  "users": [
    {
      "username": "michaelScott",
      "email": "michael_scott@dunder.com",
      "password": "6EU9kypK",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 1
    },
    {
      "username": "jimHalpert",
      "email": "jim_halpert@dunder.com",
      "password": "W4uriNy",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 2
    },
    {
      "username": "pamBeesly",
      "email": "pam_beesly@dunder.com",
      "password": "wX24hZ",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 3
    },
    {
      "username": "dwightSchrute",
      "email": "dwight_schrute@dunder.com",
      "password": "12345",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 4
    }
  ],
  "messages": [
    {
      "userId": 1,
      "message": "Name the best hot chicken in the ville, go!",
      "date": "7/17/2020 19:10:50",
      "id": 1
    },
    {
      "userId": 2,
      "message": "Hattie B's all the way!",
      "date": "7/17/2020 19:20:00",
      "id": 2
    },
    {
      "userId": 3,
      "message": "Prince's hands down!",
      "date": "7/17/2020 19:34:20",
      "id": 3
    },
    {
      "userId": 1,
      "message": "🔥🐓🍗",
      "date": "7/17/2020 19:48:27",
      "id": 4
    },
    {
      "userId": 4,
      "message": "Hello",
      "date": "7/18/2020 10:30:08",
      "id": 5
    },
    {
      "userId": 4,
      "message": "Dwight here, I'm not your friend Michael",
      "date": "7/18/2020 10:30:08",
      "id": 6
    }
  ]
